addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -186,6 +189,36 @@ MockServer.include({
         });
         return messages;
     },
+    /**
+     * Simulate the 'message_post' Python method
+     *
+     * @private
+     * @return {Object[]}
+     */
+    _mockMessagePost: function (args) {
+        const {
+            args: [res_id],
+            model: res_model,
+            kwargs: postData,
+        } = args;
+        const model = this.data['mail.message'];
+        const modelFieldNames = Object.keys(model.fields);
+        const record = {};
+        for (const fieldName in postData) {
+            if (modelFieldNames.includes(fieldName)) {
+                record[fieldName] = postData[fieldName];
+            }
+        }
+        const amountOfRecords = this.data['mail.message'].records.length;
+        const id = amountOfRecords + 1;
```

Comment:
Don't rely on amount of `mail.message` records for computing next id.
Instead take increment of current max id (and if none, start with id 1).